apollo/lib/schemas/user.py

```diff
@@ -25,13 +32,30 @@ def username_must_be_unique(cls, value, **kwargs):
     @validator('password')
     @classmethod
     def no_whitespace_in_password(cls, value):
-        if ' ' in value:
-            raise ValueError("password can't contain whitespaces")
-
-        return value
+        return check_for_whitespace(value)
 
 
 class UserSchema(ORMBase):
     id: uuid.UUID
     username: str
+    has_changed_initial_password: bool
     role: Optional[RoleSchema]
+
+
+class UpdateUserSchema(BaseModel):
+    old_password: str
+    password: constr(min_length=8, strip_whitespace=True)
+    password_confirm: constr(min_length=8, strip_whitespace=True)
+
+    @validator('password')
+    @classmethod
+    def no_whitespace_in_password(cls, value):
+        return check_for_whitespace(value)
+
+    @validator('password_confirm')
```

Comment:
Old password should also be asserted as required and probably ideally validated here.